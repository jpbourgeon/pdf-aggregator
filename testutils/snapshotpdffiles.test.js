const { resolve } = require('app-root-path');
const snapshotPdfFiles = require('./snapshotpdffiles');

const testbed = resolve('testutils/__testbed__/snapshotpdffiles');

describe('The test utility function snapshotPdfFiles', () => {
  it('should return an empty array from a folder without pdf files', async () => {
    expect.assertions(2);
    const folderSnapshot = await snapshotPdfFiles(testbed)
    // eslint-disable-next-line no-console
      .catch(e => console.log(`snapshotPdfFiles > snapshotPdfFiles test 1: ${e.message}`));
    expect(folderSnapshot).toHaveLength(0);
    expect(folderSnapshot).toMatchSnapshot();
  });

  it('should snapshotPdfFiles a folder with a single pdf file into an array with one element', async () => {
    expect.assertions(2);
    const folderSnapshot = await snapshotPdfFiles(`${testbed}/onepdffile`)
    // eslint-disable-next-line no-console
      .catch(e => console.log(`snapshotPdfFiles > snapshotPdfFiles test 2: ${e.message}`));
    expect(folderSnapshot).toHaveLength(1);
    expect(folderSnapshot).toMatchSnapshot();
  });

  it('should snapshotPdfFiles a folder with multiple pdf files into an array', async () => {
    expect.assertions(2);
    const folderSnapshot = await snapshotPdfFiles(`${testbed}/manypdffiles`)
    // eslint-disable-next-line no-console
      .catch(e => console.log(`snapshotPdfFiles > snapshotPdfFiles test 3: ${e.message}`));
    expect(folderSnapshot).toHaveLength(3);
    expect(folderSnapshot).toMatchSnapshot();
  });
});
