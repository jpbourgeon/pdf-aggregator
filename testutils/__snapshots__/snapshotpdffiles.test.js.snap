// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test utility function snapshotPdfFiles should return an empty array from a folder without pdf files 1`] = `Array []`;

exports[`The test utility function snapshotPdfFiles should snapshotPdfFiles a folder with a single pdf file into an array with one element 1`] = `
Array [
  "fSBPshpGgJUeUEPU957Ddwr6Eoc=",
]
`;

exports[`The test utility function snapshotPdfFiles should snapshotPdfFiles a folder with multiple pdf files into an array 1`] = `
Array [
  "1hF12PEM24LRd1M1VOyGq7K1YfI=",
  "fRlatOrKKEs4Dpy5Pzt3woQojIw=",
  "fSBPshpGgJUeUEPU957Ddwr6Eoc=",
]
`;
